### PR TITLE
OPAL-160: removal from android client

### PR DESCRIPTION
**Pull request by arnikaeva**

### CODE REVIEW

File: app/src/main/java/com/greenfox/kalendaryo/http/backend/BackendApi.java [hunk 1]
```diff
@@ -26,4 +28,8 @@
     @GET("calendar")
     Call<GetKalendarListResponse> getCalendar(@Header("X-Client-Token") String clientToken);
 
+    @DELETE("{id}/deletecalendar")
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/http/backend/BackendApi.java]: endpoint name, could be /calendars/{id} with delete action

File: app/src/main/java/com/greenfox/kalendaryo/models/responses/GetKalendarResponse.java [hunk 2]
```diff
@@ -10,6 +10,7 @@
     private String outputGoogleAuthId;
     private String outputCalendarId;
     private List<String> inputGoogleCalendars;
+    private long id;
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/models/responses/GetKalendarResponse.java]: order, id should be the first field

File: app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java [hunk 3]
```diff
@@ -53,20 +58,30 @@ public int getItemCount() {
         return getKalendarResponses.size();
     }
 
-    public class ViewHolder extends RecyclerView.ViewHolder {
+    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java]: is this really needed? Check without this

File: app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java [hunk 4]
```diff
@@ -53,20 +58,30 @@ public int getItemCount() {
...
+        public void onClick(View view) {
+        }
     }
     public interface ListChange {
         void saveCalendar(String calendarTitle);
         void removeCalendar(String calderTitle);
     }
+
+    public void removeAt(int position) {
+        getKalendarResponses.remove(position);
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java]: getKalendarResponses name should be refactored

File: app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java [hunk 5]
```diff
@@ -53,20 +58,30 @@ public int getItemCount() {
...
+        notifyItemRemoved(position);
+        notifyItemChanged(0,getKalendarResponses.size());
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java]: check if this is needed

File: app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java [hunk 6]
```diff
@@ -41,17 +55,63 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+                        String clientToken = kalPref.clientToken();
+
+                        List<GetKalendarResponse> kalendarList = adapter.getGetKalendarResponses();
+                        GetKalendarResponse kalendarToDelete = kalendarList.get(position);
+                        long deleteId = kalendarToDelete.getId();
+
+                        backendApi.deleteKalendar(clientToken, deleteId).enqueue(new Callback<PostKalendarResponse>() {
+                            @Override
+                            public void onResponse(Call<PostKalendarResponse> call, Response<PostKalendarResponse> response) {
+                                PostKalendarResponse postKalendarResponse = response.body();
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java]: only remove the item from the view, if the request was succesful `adapter.removeAt(position);`

File: app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java [hunk 7]
```diff
@@ -41,17 +55,63 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+                            }
+
+                            @Override
+                            public void onFailure(Call<PostKalendarResponse> call, Throwable t) {
+                                t.printStackTrace();
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java]: if it was not succesful, show a toast to the user about the fact

File: app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java (hunk 6 above)
Comment (arnikaeva) [reply app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java]: and show a toast that it was succesful

File: app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java [hunk 8]
```diff
@@ -41,17 +55,63 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+
         recyclerView.addItemDecoration(dividerItemDecoration);
         kalPref = new KalPref(this.getContext());
         getKalendarResponse(kalPref.clientToken());
 
+        recyclerView.addOnItemTouchListener(new RecyclerTouchListener(getActivity(), recyclerView, new ClickListener() {
+            @Override
+            public void onLongClick(View view, int position) {
+                AlertDialog.Builder alert = new AlertDialog.Builder(getActivity());
+                alert.setTitle("Hey mate!");
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java]: refactor

File: app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java [hunk 9]
```diff
@@ -76,4 +136,45 @@ public void onFailure(Call<GetKalendarListResponse> call, Throwable t) {
...
+    }
+
+    class RecyclerTouchListener implements RecyclerView.OnItemTouchListener{
+
+        private ClickListener clickListener;
+        private GestureDetector gestureDetector;
+
+        public RecyclerTouchListener(Context context, final RecyclerView recyclerView,
+                                     final ClickListener clickListener){
+            this.clickListener = clickListener;
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java]: discuss

File: app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java [hunk 10]
```diff
@@ -1,37 +1,62 @@
...
     }
-    public void addKalendarResponse(List<GetKalendarResponse> newGetKalendarResponses) {
-        this.getKalendarResponses.addAll(newGetKalendarResponses);
+
+    public List<GetKalendarResponse> getKalendarResponses() {
+        return kalendarResponses;
+    }
+
+    public void addKalendarResponse(List<GetKalendarResponse> kalendarResponses) {
+        this.kalendarResponses.addAll(kalendarResponses);
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java]: addAll will append the new elements to the existing list. Is this what we want?

File: app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java [hunk 11]
```diff
@@ -43,14 +68,78 @@ public void addKalendarResponse(List<GetKalendarResponse> newGetKalendarResponse
...
+        AlertDialog.Builder alert = new AlertDialog.Builder(view.getContext());
+        alert.setTitle("Delete Kalendar");
+        alert.setMessage("Are you sure to delete the kalendar?");
+        alert.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialogInterface, int i) {
+                kalPref = new KalPref(view.getContext());
+                String clientToken = kalPref.clientToken();
+
+                backendApi.deleteKalendar(clientToken, kalendarIdToDelete(position)).enqueue(new Callback<Void>() {
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java]: do not put a method call as a method parameter, save the kalendarID into a variable before the backend api call

**Review by arnikaeva — COMMENTED** [review]
minor changes

File: app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java (hunk 10 above)
Comment (markaron) [reply app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java]: This function was already in the code. Not sure of the answer.

**Review by sznemeth3 — DISMISSED** [review]

**Review by bekobarna — DISMISSED** [review]

**Review by markaron — DISMISSED** [review]

File: app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java [hunk 12]
```diff
@@ -43,14 +63,80 @@ public void addKalendarResponse(List<GetKalendarResponse> newGetKalendarResponse
 
     @Override
     public void onBindViewHolder(KalendarAdapter.ViewHolder holder, int position) {
-        GetKalendarResponse getKalendarResponse = getKalendarResponses.get(position);
+        DaggerApiComponent.builder().build().inject(this);
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                popUpWindow(view, position);
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java]: would use more meaningful name, like `deleteKalendarPopUp`

**Review by arnikaeva — COMMENTED** [review]
minor refactor

File: app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java (hunk 12 above)
Comment (markaron) [reply app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java]: done

**Review by sznemeth3 — APPROVED** [review]

**Review by bekobarna — APPROVED** [review]

**Pull request merged**